stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt

```diff
@@ -0,0 +1,35 @@
+package io.getstream.chat.android.ui.messages.adapter.viewholder.decorator
+
+import android.widget.TextView
+import com.getstream.sdk.chat.adapter.MessageListItem
+import com.getstream.sdk.chat.utils.DateFormatter
+import com.getstream.sdk.chat.utils.formatTime
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.utils.extensions.getCreatedAtOrThrow
+
+internal class TimeDecorator(private val dateFormatter: DateFormatter) : BaseDecorator() {
+
+    override fun decoratePlainTextMessage(viewHolder: MessagePlainTextViewHolder, data: MessageListItem.MessageItem) {
+        setupTime(viewHolder.binding.tvTime, data)
+    }
+
+    override fun decoratePlainTextWithMediaAttachmentsMessage(
+        viewHolder: PlainTextWithMediaAttachmentsViewHolder,
+        data: MessageListItem.MessageItem
+    ) {
+        setupTime(viewHolder.binding.tvTime, data)
+    }
+
+    override fun decorateOnlyMediaAttachmentsMessage(
+        viewHolder: OnlyMediaAttachmentsViewHolder,
+        data: MessageListItem.MessageItem
+    ) {
+        setupTime(viewHolder.binding.tvTime, data)
+    }
+
+    private fun setupTime(textView: TextView, data: MessageListItem.MessageItem) {
+        textView.text = dateFormatter.formatTime(data.message.getCreatedAtOrThrow())
```

Comment:
Do we really want to throw an exception here? For example, component browser screen crashes because mocked messages don't contain such a field

Reply:
CreatedAt sholdn't be never null (it is set by server or our local mechanism), if it is null, message has wrong data (Maybe because dev has set as null it).
Should we allow null there? :thinking: